java/ql/lib/ext/java.nio.file.model.yml

```diff
@@ -19,18 +19,47 @@ extensions:
       - ["java.nio.file", "Files", False, "write", "", "", "Argument[1]", "write-file", "manual"]
       - ["java.nio.file", "Files", False, "writeString", "", "", "Argument[0]", "create-file", "manual"]
       - ["java.nio.file", "Files", False, "writeString", "", "", "Argument[1]", "write-file", "manual"]
+      - ["java.nio.file", "Files", True, "createDirectories", "(Path,FileAttribute[])", "", "Argument[0]", "create-file", "ai-generated"]
+      - ["java.nio.file", "Files", True, "createDirectory", "(Path,FileAttribute[])", "", "Argument[0]", "create-file", "ai-generated"]
+      - ["java.nio.file", "Files", True, "createFile", "(Path,FileAttribute[])", "", "Argument[0]", "create-file", "ai-generated"]
+      - ["java.nio.file", "Files", True, "createTempFile", "(Path,String,String,FileAttribute[])", "", "Argument[0]", "create-file", "ai-generated"]
```

Comment:
> These four models appear to be duplicates of the models at lines 8, 9, 10, and 14 above

When the signature is `""`, does that mean the model should apply to ANY signature? If so, this is true (and I've learned something new). If not, they're not duplicates.

> I believe the subtype (True/False) needs to be ignored when sorting

Can do for future PRs.

Reply:
> When the signature is "", does that mean the model should apply to ANY signature? If so, this is true (and I've learned something new). If not, they're not duplicates.

Yes, `""` as the signature applies to any signature. Is the ML model currently able to exclude this case? Or does this FP pattern need to be reported to @tiferet?

Reply:
The modeling excludes all nodes `n` surfaced by `sinkNode(n, "create-file")`. Does `sinkNode` treat a signature of `""` as expected, surfacing the relevant argument for this method regardless of method signature?

Reply:
But we use `not exists(string kind | sinkNode(n, kind)` to filter out sink candidates, so `n` should never end up as a candidate (and therefore never be surfaced in the AI-generated YML file, if it was selected by `sinkNode(n, "create-file")`. Maybe I have a bug in my code somewhere 🤔. @kaeluka do you think you can take a 👀 ? [Here](https://github.com/github/codeql/blob/68133cb8ca3f2859bca4e8d69c31404c943c2c82/java/ql/experimental/adaptivethreatmodeling/lib/experimental/adaptivethreatmodeling/EndpointCharacteristics.qll#L254) and [here](https://github.com/github/codeql/blob/68133cb8ca3f2859bca4e8d69c31404c943c2c82/java/ql/experimental/adaptivethreatmodeling/src/ExtractSinkCandidates.ql#L67) 🙏

Reply:
I'll take a look on Monday! These bugs are really tricky to find because it may well be that the bug has been fixed already, we're just looking at data obtained using really old candidates :D

Reply:
> I'll take a look on Monday! These bugs are really tricky to find because it may well be that the bug has been fixed already, we're just looking at data obtained using really old candidates :D

This is true! How about you take a quick look, and if you don't find something obvious, let's assume there's no bug. If the problem recurs, we'll open an issue and dig in more thoroughly. Sound good?

Reply:
(resolving this because the original comment has been addressed)